src/swarm-lang/Swarm/Language/Parser/Term.hs

```diff
@@ -107,6 +107,7 @@ parseTermAtom2 =
           <*> pure Nothing
           <*> (optional (symbol ";") *> (parseTerm <|> (eof $> sNoop)))
         <|> SRcd <$> brackets (parseRecord (optional (symbol "=" *> parseTerm)))
+        <|> TType <$> (symbol "@" *> parseTypeAtom)
```

Comment:
🤩

Reply:
Yeah, doing it this more general way turned out to be way nicer than making `read` a special syntax form.  It actually wasn't even that hard since we already had types-as-values for the implementation of `read`.